### PR TITLE
Update to Apache Iceberg 1.4.0

### PR DESCRIPTION
**Pull request by Fokko**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -176,7 +176,7 @@
         <dep.kafka-clients.version>3.3.2</dep.kafka-clients.version>
         <dep.casandra.version>4.17.0</dep.casandra.version>
         <dep.minio.version>8.5.5</dep.minio.version>
-        <dep.iceberg.version>1.3.1</dep.iceberg.version>
+        <dep.iceberg.version>1.4.0</dep.iceberg.version>
```
Comment (ajantha-bhat) [inline pom.xml:179]: we also need to bump the matching nessie version while bumping the Iceberg version. 

Please refer the previous Iceberg version bump PR
https://github.com/trinodb/trino/pull/17715/files

Comment (Fokko) [reply pom.xml:179]: The CI is failing on this. I think it is best to do the Nessie bump in a separate PR.

Comment (ajantha-bhat) [reply pom.xml:179]: But they are dependent on each other. Sometimes newer version of Nessie may not work with the Nessie client in older version of the Iceberg. Hence, we used to bump them together.

**Comment by aokolnychyi** [comment]
I think tests and potentially logic in Trino has to be adapted to consume Iceberg 1.4. In particular, there is a behavior change w.r.t. how files are being cleaned up. [Here](https://github.com/apache/iceberg/pull/8397#issue-1867885792) is a PR with more details. In short, Iceberg introduced a strict cleanup mode and uses it by default to prevent corrupting tables when it is not clear whether the commit is successful.

**Comment by ajantha-bhat** [comment]
@Fokko: Looks like only two checks ran after bumping the pom to use new RC. We might have to rebase to re-trigger the build.

**Comment by Fokko** [comment]
@ajantha-bhat Thanks, I've resolved the conflicts and all the tests are running again

File: pom.xml [hunk 2]
```diff
@@ -174,7 +174,7 @@
         <dep.kafka-clients.version>3.3.2</dep.kafka-clients.version>
         <dep.casandra.version>4.17.0</dep.casandra.version>
         <dep.minio.version>8.5.6</dep.minio.version>
-        <dep.iceberg.version>1.3.1</dep.iceberg.version>
+        <dep.iceberg.version>1.4.0</dep.iceberg.version>
```
Comment (ebyhr) [inline pom.xml:177]: Could you remove TrinoParquetUtil class? We introduced the class temporarily as workaround of https://github.com/apache/iceberg/pull/8559.

**Comment by ajantha-bhat** [comment]
@Fokko: Are you still working on this?

File: pom.xml (hunk 2 above)
Comment (Fokko) [reply pom.xml:179]: Thanks for reminding me, and certainly, less is more.

**Comment by Fokko** [comment]
@ajantha-bhat Fell of my radar, but I've cleaned up the PR 👍

**Review by nastra — APPROVED** [review]
changes LGTM, but looks like there are some CI failures

**Comment by findinpath** [comment]
Probably superseded by https://github.com/trinodb/trino/pull/19434

**Comment by Fokko** [comment]
@findinpath Yes, thank you 👍

**Pull request closed without merging**